pkg/monitor/controller.go

```diff
@@ -303,6 +311,9 @@ func (c *controllerMonitor) processNextExternalNodeWorkItem() bool {
 }
 
 func (c *controllerMonitor) syncNode(key string) error {
+	if !c.AgentInfoCRDInstalled() {
+		return nil
+	}
```

Comment:
I think we should just not run these goroutines instead of check the API's existence on every Node event?

Reply:
Is it possible that at time 1 the antreaagentinfo CRD is not deployed, but at time2 the CRD is applied. If we only have a single check at the initial run, the case is not supported.

Reply:
The code doesn't automatically call `deleteStaleAgentCRDs` and `syncNode` when the CRD is added after starting controller, right? Then it would just lead to an incomplete situation that some Nodes are handled while others are not, so a restart is needed anyway. Besides, there seems no reason to apply CRD after running controller? Could other features consuming CRD work correctly? But anyway I feel not worth to add repeated API calls to partially support a scenario that we don't have yet. 

And I think the point of the PR is to avoid adding a configuration that controls whether to run the AntreaAgentInfo controller, by auto-discovering it. If the auto-discovery becomes more complicated and even double APIs call of normal mode, it may be not worth.

Reply:
removed.